src/cargo/util/toml/mod.rs

```diff
@@ -2151,6 +2164,40 @@ fn to_dependency_source_id<P: ResolveToPath + Clone>(
     }
 }
 
+pub(crate) fn lookup_path_base(
+    base: &str,
+    gctx: &GlobalContext,
+    manifest_root: &Path,
+) -> CargoResult<PathBuf> {
+    if !gctx.cli_unstable().path_bases {
+        bail!("usage of path bases requires `-Z path-bases`");
+    }
+
+    // Look up the relevant base in the Config and use that as the root.
+    if let Some(path_bases) =
+        gctx.get::<Option<ConfigRelativePath>>(&format!("path-bases.{base}"))?
```

Comment:
We should probably have a comment explaining why this path-injection is safe from abuse

Reply:
Done

Reply:
ok